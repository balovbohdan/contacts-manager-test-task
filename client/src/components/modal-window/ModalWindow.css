.main {
    z-index: 99999;
    position: absolute;

    width: 100vw;
    height: 100vh;

    top: 0;
    left: 0;
    right: 0;
    bottom: 0;

    display: flex;
    align-items: center;
    justify-content: center;
}

.body {
    max-width: 450px;
    width: calc(100vw - 20px);

    min-height: 200px;
    max-height: calc(100vh - 40px);

    padding: 10px;
    margin: 20px auto;

    position: relative;
    border-radius: var(--border-radius);
    background-color: var(--color-light-blue);
    box-shadow: 0 20px 35px rgba(0, 0, 0, .3), 0 15px 12px rgba(0, 0, 0, .2);

    color: var(--color-white);
}

.head {
    position: relative;
    padding-bottom: 10px;

    display: flex;
    align-items: center;
    justify-content: flex-start;
}

.closeBtn {
    position: absolute;

    top: -20px;
    right: -20px;

    width: 40px;
    height: 40px;

    display: flex;
    align-items: center;
    justify-content: center;

    cursor: pointer;
    border-radius: 50%;
    transition: var(--trans);
    background-color: var(--color-red);
    box-shadow: 0 3px 10px rgba(0, 0, 0, .3), 0 5px 10px rgba(0, 0, 0,.2);
}

.closeBtn:hover {
    transform: rotate(10deg) scale(.9);
}

.closeBtn_img {
    width: calc(100% - 20px);
    height: calc(100% - 20px);

    background-size: contain;
    background-repeat: no-repeat;
    background-position: center center;
    background-image: url(~@img/close.svg);
}