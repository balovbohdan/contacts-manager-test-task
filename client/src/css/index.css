@import url(reset.css);
@import url(fonts.css);
@import url(titles.css);
@import url(constants.css);
@import url(main.css);