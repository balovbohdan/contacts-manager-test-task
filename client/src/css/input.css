.label {
    font-size: .9rem;
}

.input {
    width: 100%;
    padding: 5px 0;
    border-bottom: 2px solid var(--color-yellow);

    font-size: 1.5rem;
    color: var(--color-white);
}

@media (max-width: 600px) {
    .input {
        font-size: 1.2rem;
    }

    .label {
        font-size: .8rem;
    }
}

@media (max-width: 400px) {
    .input {
        font-size: 1.1rem;
    }

    .label {
        font-size: .7rem;
    }
}