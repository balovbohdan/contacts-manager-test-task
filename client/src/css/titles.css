h1, h2, h3, h4, h5, h6 {
    font-size: 100%;
    font-family: var(--font-main);
}

h1 {
    font-size: 40px;
}

h2 {
    font-size: 35px;
}

h3 {
    font-size: 30px;
}

h4 {
    font-size: 25px;
}

h5 {
    font-size: 20px;
}

h6 {
    font-size: 17px;
}

@media (max-width: 550px) {
    h1 {
        font-size: 35px;
    }

    h3 {
        font-size: 25px;
    }
}

@media (max-width: 400px) {
    h1 {
        font-size: 30px;
    }

    h3 {
        font-size: 20px;
    }
}