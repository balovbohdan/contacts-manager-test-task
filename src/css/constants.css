:root {
    --color-red: #FB4B4E;
    --color-blue: #16324F;
    --color-white: #F3F7F0;
    --color-yellow: #FDE74C;
    --color-light-blue: #236098;

    --font-main: Roboto, serif;
    --font-marker: 'Permanent Marker', serif;

    --border-radius: 3px;
    --header-height: 60px;
    --trans: all .12s ease-in;
}