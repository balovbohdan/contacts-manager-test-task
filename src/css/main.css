body {
    background: var(--color-light-blue);
}