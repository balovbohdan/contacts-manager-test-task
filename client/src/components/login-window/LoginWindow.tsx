import * as React from 'react';

import {ModalWindow} from '@components/modal-window';

type Props = {
    close:()=>void;
};

export const LoginWindow = ({close}:Props) =>
    <ModalWindow title='Login' close={close}>
        <h1>Login Window Body</h1>
    </ModalWindow>;