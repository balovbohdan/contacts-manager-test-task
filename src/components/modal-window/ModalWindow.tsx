import * as React from 'react';

import {Fader} from '@components/fader';

import css from './ModalWindow.css';

type Props = {
    close:Close;
    title:string;
    children:React.ReactNode;
};

type HeadProps = {
    close:Close;
    title:string;
};

type CloseBtnProps = {
    close:Close;
};

type Close = ()=>void;

export const ModalWindow = ({close, title, children}:Props) => {
    const onWrapperClick = createOnWrapperClick(close);

    return (
        <Fader>
            <div className={css.main} onClick={onWrapperClick}>
                <div className={css.body}>
                    <Head close={close} title={title}/>
                    {children}
                </div>
            </div>
        </Fader>
    );
};

const Head = ({close, title}:HeadProps) =>
    <div className={css.head}>
        <h5>{title}</h5>
        <CloseBtn close={close}/>
    </div>;

const CloseBtn = ({close}:CloseBtnProps) =>
    <div onClick={close} className={css.closeBtn}>
        <div className={css.closeBtn_img}/>
    </div>;

const createOnWrapperClick = (close:Close) =>
    ({target}) => {
        const isMain = target.classList.contains(css.main);

        if (isMain) close();
    };