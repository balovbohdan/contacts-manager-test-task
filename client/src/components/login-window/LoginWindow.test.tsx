import * as React from 'react';
import {mount, shallow} from 'enzyme';

import {LoginWindow} from './LoginWindow';
import * as css from '../modal-window/ModalWindow.css';

describe('LoginWindow', () => {
    it('Should match snapshot', () => {
        const close = jest.fn();
        const window = shallow(<LoginWindow close={close}/>);

        expect(window).toMatchSnapshot();
    });

    it('Should react on close btn click', () => {
        const close = jest.fn();
        const window = mount(<LoginWindow close={close}/>);

        window.find(`.${css.closeBtn}`).simulate('click');
        expect(close).toBeCalledTimes(1);
    });

    it('Should close after empty space click', () => {
        const close = jest.fn();
        const window = mount(<LoginWindow close={close}/>);

        window.find(`.${css.main}`).at(0).simulate('click');
        expect(close).toBeCalledTimes(1);
    });
});