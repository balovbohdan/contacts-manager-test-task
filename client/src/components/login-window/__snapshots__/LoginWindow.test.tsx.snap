// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginWindow Should match snapshot 1`] = `
<Component
  close={[MockFunction]}
  title="Login"
>
  <h1>
    Login Window Body
  </h1>
</Component>
`;
